src/dynamic-breakpoints.php

```diff
@@ -131,16 +131,42 @@ public function adjust_breakpoints( $css ) {
 			$new_tablet = $breakpoints['tablet'];
 			$new_mobile = $breakpoints['mobile'];
 
-			if ( ! empty( $new_tablet ) ) {
-				$css = preg_replace( "/(@media[^{]+)width:\s*1024px/", "$1width:" . $new_tablet . "px", $css );
-				$css = preg_replace( "/(@media[^{]+)width:\s*1023px/", "$1width:" . ( $new_tablet - 1 ) . "px", $css );
-			}
 
-			if ( ! empty( $new_mobile ) ) {
-				$css = preg_replace( "/(@media[^{]+)width:\s*768px/", "$1width:" . $new_mobile . "px", $css );
-				$css = preg_replace( "/(@media[^{]+)width:\s*767px/", "$1width:" . ( $new_mobile - 1 ). "px", $css );
+			if ( $new_tablet === '1024' && $new_mobile === '768' ) {
+				return $css;
 			}
 
+			// Check if css still have the default tablet and mobile breakpoints
+			$has_default_tablet = strpos( $css, 'width:1024px)' ) !== false || strpos( $css, 'width:1023px)' ) !== false;
+			$has_default_mobile = strpos( $css, 'width:768px)' ) !== false || strpos( $css, 'width:767px)' ) !== false;
+
+            if ( ! empty( $new_tablet ) && $has_default_tablet ) {
```

Comment:
I suggest placing the `stripos` stuff inside the `if` statement since those aren't really necessary if there's no `$new_tablet` provided. Please also do the same with mobile

```
if ( ! empty( $new_tablet ) ) {
	// Check if css still have the default tablet and mobile breakpoints
	$has_default_tablet = strpos( $css, 'width:1024px)' ) !== false || strpos( $css, 'width:1023px)' ) !== false;
	$has_new_tablet = strpos( $css, 'width:' . $new_tablet . 'px)' ) !== false;
	$has_new_tablet_minus_1 = strpos( $css, 'width:' . ( $new_tablet - 1 ) . 'px)' ) !== false;
	if ( $has_default_tablet && ! $has_new_tablet  ) {
```